### ISSUE TITLE
feat(rpc): Add encode/decode to state diff field 

### ISSUE THREAD

**Issue by antiyro** [keep-starknet-strange/madara#1327]
### feat: Implement Encode/Decode Trait for StateDiff Field

A new `StateDiff` field has been introduced in `InvokeTransactionTrace`, `DeployAccountTransactionTrace`, `L1HandlerTransactionTrace`, and `DeclareTransactionTrace` to address issue #1313. 

Currently, this field lacks the implementation of the encode/decode trait, which is essential for its correct integration into the respective structures. This deficiency is highlighted by the following error:

```error
the trait bound `[starknet_core::types::StateDiff]: Encode` is not satisfied
```

To resolve this, a comprehensive implementation of the `StateDiff` with encodable and decodable components is necessary. This implementation should adhere to the established schema in [starknet-rs](https://github.com/xJonathanLEI/starknet-rs/blob/6cadb198644cf5e942187896c2e6bb3ecc9edc6e/starknet-core/src/types/codegen.rs#L1684), thereby aligning with the JSON-RPC specifications.

**Comment by antiyro** [comment]
I'm taking this one

**Comment by antiyro** [comment]
Closing this issue as it is solved here: https://github.com/keep-starknet-strange/madara/pull/1313/commits/5641578b97507aeb2b234669149cc3c887321919 and about to be merged here #1313

**Comment by antiyro** [comment]
Note that this issue is linked to this one: https://github.com/keep-starknet-strange/madara/issues/1331
